strings/slices/slices.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package slices defines various functions useful with slices of string type.
+// The goal is to be as close as possible to
+// https://github.com/golang/go/issues/45955. Ideal would be if just
+// the import can be updated if the "slices" package becomes standard.
+package slices
+
+// func Filter[T any](d, s []T, keep func(T) bool) []T
+
+// Filter returns a new slice with the strings from the passed slice that are valid.
+
+// Filter appends to d each element e of s for which keep(e) returns true.
+// It returns the modified d. d may be s[:0], in which case the kept
+// elements will be stored in the same slice.
+// if the slices overlap in some other way, the results are unspecified.
+// To create a new slice with the filtered results, pass nil for d.
+func Filter(d, s []string, keep func(string) bool) []string {
+	if d == nil {
+		d = []string{}
+	}
+	for _, n := range s {
+		if keep(n) {
+			d = append(d, n)
+		}
+	}
+	return d
+}
+
+// Contains reports whether v is present in s.
+func Contains(s []string, v string) bool {
```

Comment:
if both are empty it should return true?

Reply:
Good question. Should it? I am unsure. Maybe a good question in https://github.com/golang/go/issues/45955

Reply:
https://github.com/golang/go/issues/45955#issuecomment-873348807

Reply:
:+1:

Reply:
Answer: false